src/api/middleware/application.js

```diff
@@ -0,0 +1,19 @@
+import CompanyApplication, { CompanyApplicationRules } from "../../models/CompanyApplication.js";
+import { APIError, ErrorTypes } from "./errorHandler.js";
+import { StatusCodes as HTTPStatus } from "http-status-codes/build/cjs/status-codes.js";
+import { RECOVERY_LINK_EXPIRATION } from "../../models/constants/ApplicationStatus.js";
+
+export const exceededCreationTimeLimit = async (email) => {
+    const cursor = await CompanyApplication.findOne({ email, isVerified: false }).exec();
+    if (cursor !== null && Date.now() - cursor.submittedAt < RECOVERY_LINK_EXPIRATION * 1000) {
```

Comment:
The logic here is quite confusing. I suggest reversing

```suggestion
    if (application !== null && Date.now() < application.submittedAt + RECOVERY_LINK_EXPIRATION * SECOND_IN_MS) {
```

Reply:
Isn't it possible to use this without the submittedAt field?

Reply:
Perhaps you could inverse the verification and check for errors or incorrect state, signaling it with a corresponding error. I don't know if I made myself clear.

Reply:
@DoStini ?